lapis/src/main/kotlin/org/genspectrum/lapis/response/CsvWriter.kt

```diff
@@ -44,6 +45,63 @@ class CsvWriter {
     }
 }
 
+/**
+ * The IANA TSV Writer implements TSV output according to this spec:
+ * https://www.iana.org/assignments/media-types/text/tab-separated-values
+ *
+ * It escapes Tabs and newlines in cell values (instead of quoting the cell content).
+ * Rudimentary TSV parsing with simple splitting will not handle quoted cell values,
+ * and thus can cause issues, this is why this implementation was added.
+ *
+ * The class is called TSV writer, but it's still possible to set other delimiters.
+ *
+ * Newlines will be escaped as \n, tabs as \t, and other delimiters as \<delimiter>.
+ */
+@Component
+class IanaTsvWriter {
+    fun write(
+        appendable: Appendable,
+        includeHeaders: Boolean,
+        data: RecordCollection<*>,
+        delimiter: Delimiter,
+    ) {
+        if (includeHeaders) {
+            writeRow(appendable, data.getHeader(), delimiter)
+        }
+        data.getCsvRecords().forEach { record ->
+            writeRow(appendable, record.map { it.orEmpty() }, delimiter)
+        }
+    }
+
+    private fun writeRow(
+        appendable: Appendable,
+        cells: List<String>,
+        delimiter: Delimiter,
+    ) {
+        val row = getRow(cells, delimiter)
+        appendable.appendLine(row)
+        if (appendable is Flushable) {
+            (appendable as Flushable).flush()
+        }
+    }
+
+    private fun getRow(
+        record: List<String>,
+        delimiter: Delimiter,
+    ): String =
+        record.joinToString(separator = delimiter.value.toString()) { cell ->
+            cell
+                .replace("\n", "\\n")
+                .let { v ->
+                    if (delimiter.value == '\t') {
+                        v.replace("\t", "\\t")
+                    } else {
+                        v.replace(delimiter.value.toString(), "\\${delimiter.value}")
+                    }
```

Comment:
Why do we need two cases here? Wouldn't the `else` case be enough?

Reply:
no because then it will replace \t with \\\t unfortunately